### PR TITLE
[CARBONDATA-1008] use MetastoreListener to sync schema between spark and hive

### PR DESCRIPTION
**Pull request by cenyuhai**
use MetastoreListener to sync schema between spark and hive

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by cenyuhai** [comment]
ok to test

**Comment by cenyuhai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3272/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/677/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/678/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3273/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/679/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3274/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/680/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3275/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/687/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3282/

**Comment by cenyuhai** [comment]
@chenliang613 @jackylk

File: integration/hive/pom.xml [hunk 1]
```diff
@@ -136,6 +142,12 @@
             <groupId>org.apache.carbondata</groupId>
             <artifactId>carbondata-hadoop</artifactId>
             <version>${project.version}</version>
+            <exclusions>
+              <exclusion>
+                <groupId>org.apache.spark</groupId>
+                <artifactId>spark-sql_2.10</artifactId>
```
Comment (anubhav100) [inline integration/hive/pom.xml]: Why spark-sql-2.10 required

Comment (cenyuhai) [inline integration/hive/pom.xml]: @anubhav100 It is not required, so I exclude it.

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by chenliang613** [comment]
@cenyuhai  can you provide the steps, how to verify it?

**Comment by cenyuhai** [comment]
@chenliang613 just do as the hive-guild.md said

**Comment by cenyuhai** [comment]
You can try what will happen after you create the table in spark with or without the listener.  And then use this comand "desc table1" to see the difference

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3332/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/735/

**Comment by CarbonDataQA** [comment]
SDV Build Failed with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/77/

**Comment by cenyuhai** [comment]
retest this please

**Comment by chenliang613** [comment]
@cenyuhai  
you mean:  use the listener, it is able to sync up table's meta info between spark and hive: 
first i create one table with name "table1" in spark-shell
second, i go to hive client , use "desc table1", i can find table1's meta info.  otherwise without this listener, the table1's meta info can't be gotten in hive client ?

**Comment by cenyuhai** [comment]
yes

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3338/

**Comment by CarbonDataQA** [comment]
SDV Build Success with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/81/

**Comment by cenyuhai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/741/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3342/

**Comment by cenyuhai** [comment]
@chenliang613 why it will failed？

**Comment by CarbonDataQA** [comment]
SDV Build Failed with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/84/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/744/

**Comment by chenliang613** [comment]
retest this please

**Comment by ravipesala** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3351/

**Comment by ravipesala** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/753/

**Comment by ravipesala** [comment]
SDV Build Failed with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/91/

**Comment by cenyuhai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3385/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/788/

**Comment by CarbonDataQA** [comment]
SDV Build Success with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/115/

**Comment by cenyuhai** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Failed with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/116/

**Comment by cenyuhai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/797/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3395/

**Comment by CarbonDataQA** [comment]
SDV Build Failed with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/124/

**Comment by cenyuhai** [comment]
retest this please

**Comment by chenliang613** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/177/

**Pull request closed without merging**